tsfc/driver.py

```diff
@@ -142,6 +142,7 @@ def compile_integral(integral_data, form_data, prefix, parameters, interface, *,
     builder.set_coordinates(mesh)
     builder.set_cell_sizes(mesh)
     builder.set_coefficients(integral_data, form_data)
+    builder.set_constants(form_data.constants)  # FIXME: should attach constants to integral data instead
```

Comment:
This could do with a more instructive comment about why this is the case. We want to attach to integral data since we don't want to be passing constants to kernels that don't need them. This is also probably a `TODO` instead of a `FIXME`.

Aside: I would love to have this refactored at some point so we don't need to distinguish here between coefficients and constants. At the point where we care about them as kernel arguments they are equivalent.